### ISSUE TITLE
Open in a temporary container always, unless I hold ctrl

### ISSUE THREAD

**Issue by zdat** [stoically/temporary-containers#385]
I want to always open in a temporary container, unless I ctrl+click, or ctrl+middle mouse.

I haven't really got this extension figured out yet. Is what I am asking possible?

**Comment by stoically** [comment]
You could enable Automatic Mode and configure Left Mouse Click Isolation (to e.g. Different from Tab Domain) - that would mostly achieve what you're asking for; however mouse click isolation [isn't that reliable](#184), so I guess it would be a nice to allow relaxing/disabling Navigation Isolation for specific mouse clicks.

**Comment by stoically** [comment]
Ah, wait - is what you're asking for that if you're already in a temporary container and ctrl/middle-mouse click a link, that it opens in the default container (no container)?

**Comment by zdat** [comment]
Yeah sorry not having any luck with that. It just continues to open in the temporary tab no matter what I change.

> however mouse click isolation [isn't that reliable](#184)

It just seems like you linked back here?

Same when I isolate a domain, then change to Mouse Click options to _Never_, so I can still open it normally. It opens in a temporary tab no matter what I press.

Basically it just seems like the Mouse Click options don't work for me at all unless I'm not understanding.

Also there's the option "Exclude Permanent Containers". Unfortunately most of my site cookies aren't in a container. They're just stored in Firefox without one. (Currently trying to figure out how to move them all to a container, if you know a way that'd be great).

Is there any way to have this extension see which site cookies you have stored and then make those sites open normally and isolate the rest? Wouldn't mind doing it manually either but I can't figure it out tbh.

**Comment by crssi** [comment]
Just as a debate which could lead to a solution... why would you sometimes like to open outside a container and not always in a container?

**Comment by zdat** [comment]
No reason really. Ideally I'd find a quick way to move my accounts with cookies into one then just exclude it.

**Comment by crssi** [comment]
For that exact reason I am using [Containerise](https://addons.mozilla.org/firefox/addon/containerise/), which nicely complements TC. 😉 

Cheers

**Comment by zdat** [comment]
Actually guys I got it working in a way that I'm fine with for now. Sorry I should have messed around a bit more before posting.

Advanced > Ignoring requests to > then I put my sites with cookies there.

Still can't figure out the Mouse Clicks though.

I'll take some time to try and understand everything.

**Comment by stoically** [comment]
> It just seems like you linked back here?

Sorry, meant to link #184 

> Same when I isolate a domain, then change to Mouse Click options to Never, so I can still open it normally. It opens in a temporary tab no matter what I press.

"Never" just ignores the mouse click completely and doesn't _isolate_ it. And with "Automatic Mode" enabled, TC will put everything that tries to load in the default container into a temporary container automatically - only way around that, as you correctly figured out - is to put domains into "Ignoring requests to".

Basically TC currently only really covers the use-cases
- Most things in TCs and a few things in permanent containers: Automatic Mode (Hard mode: +Global Isolation)
- Only put specific things into TCs and do the rest in the default container: Per Domain Isolation

The use-case to put everything into TCs but some things into the default container isn't really covered.

**Comment by zdat** [comment]
Thanks. Well I'm reasonably content with my current setup, but is there any way that I can make a site that is set to open in TC, not open in TC using a mouse click, or any way at all?

**Comment by stoically** [comment]
You mean in case it's set to "Always open in" using "Per Domain Isolation"? No, that's either enabled or disabled.

At some point in time I'd like to refactor the whole "Isolation" thing to allow adding arbitrary rules with "source > cause > action" freely definable (kind of how [Request Control](https://addons.mozilla.org/firefox/addon/requestcontrol/) does it) - but that'd be a fair chunk of work.
